File: src/SIStorage.Service.Contract/Models/SIStorageServiceError.cs
namespace SIStorage.Service.Contract.Models;

/// <summary>
/// Defines a SIStorageService error.
/// </summary>
public sealed class SIStorageServiceError
{
    /// <summary>
    /// Error code.
    /// </summary>
    public WellKnownSIStorageServiceErrorCode ErrorCode { get; set; }
}

File: web/src/models/SIStorageServiceError.ts
import WellKnownSIStorageServiceErrorCode from "./WellKnownSIStorageServiceErrorCode";

/** Defines a SIStorageService error. */
export default class SIStorageServiceError extends Error {
    /** Error code. */
    errorCode?: WellKnownSIStorageServiceErrorCode;

    /** Error status code. */
    statusCode: number;

    constructor(message: string | undefined, statusCode: number, errorCode?: WellKnownSIStorageServiceErrorCode) {
        super(message);
        this.statusCode = statusCode;
        this.errorCode = errorCode;
    }
}